internal/services/event/participants.go

```diff
@@ -0,0 +1,87 @@
+package event
+
+import (
+	"context"
+	"errors"
+	"github.com/RobinHoodArmyHQ/robin-api/internal/env"
+	"github.com/RobinHoodArmyHQ/robin-api/internal/repositories/event"
+	"github.com/RobinHoodArmyHQ/robin-api/internal/repositories/participants"
+	"github.com/RobinHoodArmyHQ/robin-api/internal/repositories/user"
+	"github.com/RobinHoodArmyHQ/robin-api/models"
+	"github.com/RobinHoodArmyHQ/robin-api/pkg/ctxmeta"
+	"github.com/RobinHoodArmyHQ/robin-api/pkg/nanoid"
+)
+
+func (i *impl) MarkEventInterested(ctx context.Context, eventID nanoid.NanoID) error {
+	userID := ctxmeta.GetUser(ctx)
+	ev := env.FromContext(ctx)
+
+	userResp, err := ev.UserRepository.GetUser(&user.GetUserRequest{UserID: userID})
+	if err != nil {
+		return err
+	}
+
+	if userResp == nil || userResp.User == nil {
+		return errors.New("user not found")
+	}
+
+	eventResp, err := ev.EventRepository.GetEvent(&event.GetEventRequest{EventID: eventID})
+	if err != nil {
+		return err
+	}
+
+	if eventResp == nil || eventResp.Event == nil {
+		return errors.New("event not found")
+	}
+
+	// Check if user is already interested in the event
+	interestResp, err := ev.ParticipantsRepository.GetEventParticipant(ctx, &participants.GetEventParticipantRequest{
+		EventID: eventResp.Event.ID,
+		UserID:  userResp.User.ID,
+	})
+	if err != nil {
+		return err
+	}
+
+	if interestResp.Participant != nil {
+		return errors.New("user is already interested in this event")
+	}
+
+	// Create participant
+	participant := &models.Participant{
+		EventID: eventResp.Event.ID,
+		UserID:  userResp.User.ID,
+		Status:  models.StatusInterested,
```

Comment:
Are we planning to update participant status to `StatusCheckedIn` in checkin api? I feel checkins can be part of participants table only and if someone wants to post/announce about their checkin/attendance - we can create entry in separate table — discussion for another time I guess.

Reply:
(From figma) photo/desc are required if someone wants to check in. 
assuming `participants >> user who check-in` it makes more sense to separate it, since most of the check-in required field will be empty if we use single table

Reply:
makes sense, can keep separate but should we add status transition in checking api?

Reply:
yup, we have checkin status in this table - will add that in next pr